File: ollama/codigo-final/mandelbrot/llama3.1-8b/fewshot/temp_0/mandelbrot.java
import java.io.OutputStream;
import java.io.PrintStream;

public class Mandelbrot {
    public static void main(String[] args) throws Exception {
        int n = Integer.parseInt(args[0]);
        int iterations = 50;
        double lim = 2.0 * 2.0;

        // Validate against reference.pbm for N=200
        if (n == 200) {
            System.out.println("Validating output...");
            compareOutput(n);
        }

        PrintStream out = new PrintStream(System.out);

        // Output PBM P4 header
        out.print("P4\n");
        out.printf("%d %d\n", n, n);
        out.print("8\n");

        int[] pixels = new int[n * n];

        for (int y = 0; y < n; y++) {
            for (int x = 0; x < n; x++) {
                double realC = 2.0 * x / n - 1.5;
                double imagC = 2.0 * y / n - 1.0;

                int zr = 0, zi = 0;
                for (int i = 0; i < iterations && zr * zr + zi * zi <= lim; i++) {
                    double temp = zr * zr + zi * zi;
                    if (temp > lim) break;
                    double newZi = realC * zr - imagC * zi;
                    zi = newZi;
                    zr = temp;
                }

                pixels[y * n + x] = zr * zr + zi * zi <= lim ? 1 : 0;
            }
        }

        // Pad with zeros if N not divisible by 8
        int padding = (n % 8 == 0) ? 0 : 8 - (n % 8);
        for (int i = n * n; i < n * n + padding; i++) {
            pixels[i] = 0;
        }

        // Output binary pixel data
        byte[] bytes = new byte[(n * n + padding) / 8];
        for (int i = 0; i < pixels.length; i += 8) {
            int byteValue = 0;
            for (int j = 0; j < 8 && i + j < pixels.length; j++) {
                byteValue |= (pixels[i + j] << j);
            }
            bytes[(i / 8)] = (byte) byteValue;
        }

        out.write(bytes);

        // Validate output
        if (n == 200) {
            System.out.println("Validation complete.");
        }
    }

    private static void compareOutput(int n) throws Exception {
        Process process = Runtime.getRuntime().exec(new String[] {"cmp", "output.pbm", "reference.pbm"});
        int exitCode = process.waitFor();
        if (exitCode != 0) {
            throw new RuntimeException("Validation failed.");
        }
    }
}

File: ollama/codigo-final/mandelbrot/llama3.1-8b/fewshot/temp_0/mandelbrot.py
import sys
from math import sqrt, pow

def mandelbrot(c, max_iter):
    z = c
    for n in range(max_iter):
        if abs(z) > 2.0:
            return n
        z = z*z + c
    return max_iter

def main():
    N = int(sys.argv[1])
    width = N
    height = N
    max_iter = 50
    lim = 2.0 * 2.0

    # Coordinate mapping
    xmin = -2.5
    xmax = 1.5
    ymin = -1.5
    ymax = 1.5

    # PBM header
    with open('mandelbrot.pbm', 'w') as f:
        f.write('P4\n')
        f.write(f'{width} {height}\n')
        f.write('8\n')

    # Binary pixel data
    pixels = bytearray(width * height)
    for y in range(height):
        for x in range(width):
            c = complex(2.0 * (x / N - 1.5), 2.0 * (y / N - 1.0))
            n = mandelbrot(c, max_iter)
            if n == max_iter:
                pixels[y * width + x] |= 0xFF
            else:
                pixels[y * width + x] &= ~0xFF

    # Pad with zeros if N not divisible by 8
    padding = (8 - (width % 8)) % 8
    pixels += bytearray(padding * height)

    # Write binary pixel data to file
    with open('mandelbrot.pbm', 'ab') as f:
        f.write(pixels)

if __name__ == '__main__':
    main()
